### PR TITLE
Colors rewrite

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Purpose of this PR is to use a uniform approach on using colors:

- Use `tint-color()` to replace `lighten()`
- Use `shade-color()` to replace `darken()`
- Use `scale-color()` to replace `color-level()` (since we don't work with levels anymore)
- Get rid of `$theme-color-interval`, the `8%` levels are hard to calculate with.
- Table variants, list item variants & alerts variants used to have different colors:
  - https://twbs-bootstrap.netlify.app/docs/5.0/content/tables/#variants
  - https://twbs-bootstrap.netlify.app/docs/5.0/components/alerts/#examples
  - https://twbs-bootstrap.netlify.app/docs/5.0/components/list-group/#contextual-classes
    vs
  - https://deploy-preview-30622--twbs-bootstrap.netlify.app/docs/5.0/content/tables/#variants
  - https://deploy-preview-30622--twbs-bootstrap.netlify.app/docs/5.0/components/alerts/#examples
  - https://deploy-preview-30622--twbs-bootstrap.netlify.app/docs/5.0/components/list-group/#contextual-classes
    (color contrast on the light variants is still an issue, but can be tackled later on to prevent this PR from being too overwhelming)

### CODE REVIEW

File: scss/_buttons.scss [hunk 1]
```diff
@@ -79,19 +82,23 @@
...
 
   &:disabled,
   &.disabled {
     color: $btn-link-disabled-color;
   }
 
-  // No need for an active state here
+  &:active {
+    border-color: transparent;
+  }
```
Comment (ffoodd) [inline scss/_buttons.scss:101]: `border`s on button seems unrelated, aren't they?

**Comment by ffoodd** [comment]
I really prefer the percentage values you're suggesting to remove `$theme-color-interval`, it's much clearer!

File: scss/_buttons.scss (hunk 1 above)
Comment (MartijnCuppens) [reply scss/_buttons.scss:101]: Yeah, I know, was just trying things out to decrease the number of unique colors.

**Comment by mdo** [comment]
Is this safe to bring back and set to ready for review for Alpha 3?

**Comment by MartijnCuppens** [comment]
Yeah, fine by me. Worked a bit on this today, but it seemed to be a bit more complex than I initially thought. I'll keep you posted

File: site/content/docs/5.0/migration.md [hunk 2]
```diff
@@ -7,6 +7,15 @@ aliases: "/migration/"
 toc: true
 ---
 
+## v5.0.0-alpha3
+
+### Colors
+
+- The color system which worked with `color-level()` and `$theme-color-interval` was removed in favor of a new color system.
+- All `lighten()` and `darken()` functions in our codebase are replaced by `tint-color()` and `shade-color()`. These functions provide will mix a color with either white or black instead of changing the lightness by a fixed amount.
```
Comment (ffoodd) [inline site/content/docs/5.0/migration.md:15]: "These functions provide will mix" seems weird, I guess "provide" should be dropped?

**Review by ffoodd — APPROVED** [review]
LGTM, really helpful changes!

**Review by mdo — APPROVED** [review]
Hell yeah, huge refactor and update, @MartijnCuppens! :shipit:

**Comment by stof** [comment]
Any reason to implement this change using black and white (essentially working in HWB) rather than using Sass's `scale-color($color, $lightness: 12%)` ?

File: scss/_functions.scss [hunk 3]
```diff
@@ -150,23 +150,23 @@ $_luminance-list: .0008 .001 .0011 .0013 .0015 .0017 .002 .0022 .0025 .0027 .003
...
+// Shade a color: mix a color with black
+@function shade-color($color, $weight) {
+  @return mix(black, $color, $weight);
 }
 
-@function shade-color($color, $level) {
-  @return mix(black, $color, $level * $theme-color-interval);
+// Scale a color:
+// Shade the color if the weight is positive, else tint it
+@function scale-color($color, $weight) {
```
Comment (stof) [inline scss/_functions.scss:166]: this function is shadowing a built-in Sass function (with the same name but a different signature), which could create lots of confusion for users importing the Bootstrap Sass files in their projects (rather than using it as modules)

Comment (ffoodd) [reply scss/_functions.scss:166]: This is already fixed in #32149.
Please open an issue or a discussion topic instead of commenting closed issue.

**Comment by miken32** [comment]
A little more guidance in the migration documentation would be helpful when 5.0 is released. Through some experimentation I found that `shift-color($color, $level * 8)` is similar to `color-level($color, $level)`.
